### PR TITLE
Jsdoc

### PR DESCRIPTION
**Pull request by lapinek**
Update code and README.md, and add JSDoc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,49 @@
...
 ## How to make it work
 
-* Copy the code from [src/scripted-debugger.js](src/scripted-debugger.js) and paste it at the bottom of your scripted decision node debugger script.
+* Copy the code from [src/scripted-debugger.js](src/scripted-debugger.js) and paste it at the bottom of your scripted decision.
+
+    > The debugger function itself, as any function statement, can be placed at and called from any place in your script.
 * In your scripted decision node script, create an array of log messages or choose a single message to be displayed in the browser.
-* Using the provided example, call the `showLogs(Object options)` function and pass in your message(s) as an option.
-* In your authentication journey/tree, add a scripted decision at a place where you want to output some content, and select the debugger script as the Script. Add "true" outcome to proceed with.
+* Using the example provided in the [src/scripted-debugger.js](src/scripted-debugger.js) file, call `showLogs(Object options)` function, and pass in your message(s) as the the `options.logs` argument.
```
Comment (lorig135) [inline README.md]: call the `showLogs(Object options)` function, and pass it in ... or do we mean paste it in ?

File: README.md [hunk 2]
```diff
@@ -1,31 +1,49 @@
...
+* Optionally, retrieve and concatenate additional logs saved in the authentication journey context—for example, in its `sharedState`.
```
Comment (lorig135) [inline README.md]: context; for example,

File: README.md (hunk 1 above)
Comment (lapinek) [reply README.md]: Pass in, as in provide (an argument for the function).

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+| [options.noPopup] | <code>boolean</code> |  | Do NOT show logs in the pop-up window. |
+| [options.noText] | <code>boolean</code> |  | Do NOT show current log in the login screen. CAUTION: the callbacks form will auto-submit if no login form callbacks are detected! In that case, make sure your journey has another stopping point (that is, a node with a callback) to avoid loops on failed login. |
+| [options.popupTitle] | <code>string</code> | <code>&quot;Debugger&quot;</code> | Window title for the pop-up window. |
+| [options.noLoggerError] | <code>boolean</code> |  | Do NOT output the logs with the `logger.error(String message)` method. If not provided or falsy, each log will be outputted with the logger object method. |
+| [options.useDebugParameter] | <code>boolean</code> |  | Require `debug` parameter in the authentication journey URL query string for displaying the log in the browser. For example, `&debug=true`. If `useDebugParameter` is `true`, and there is no `debug` parameter provided, or no value or a falsy value is provided (for example, `&debug=false`), the debugger will not be used. This will not affect the use of  the `logger.error(String message)` method. |
```
Comment (lorig135) [inline docs/README.md]: or no value,

File: src/scripted-debugger.js [hunk 4]
```diff
@@ -183,31 +192,46 @@ function showLogs (options) {
...
+            }
+
             /**
-             * Define the initial pop-up window dimensions with a JavaScript code to run on the client side.
-             *
-             * By default, it will be sized to fill the remaining space on the screen
-             * left from the (parent) login window.
-             *
-             * You can use hardcoded values instead.
+             * Create pop-up.
```
Comment (lorig135) [inline src/scripted-debugger.js]: Create a pop-up window.  ?

File: README.md (hunk 1 above)
Comment (lorig135) [reply README.md]: Thank you. Please ignore my comment.

**Review by lorig135 — APPROVED** [review]

**Pull request merged**